pygsti/construction/modelconstruction.py

```diff
@@ -1243,14 +1245,22 @@ def create_crosstalk_free_model(num_qubits, gate_names, nonstd_gate_unitaries={}
 
     if 'idle' in all_keys:
         ideal_gate = _op.StaticStandardOp('Gi', evotype)
-        idle_op = _get_error_gate('idle', ideal_gate, depolarization_strengths, stochastic_error_probs,
-                                  lindblad_error_coeffs, depolarization_parameterization,
-                                  stochastic_parameterization, lindblad_parameterization)  # 1-qubit idle op
+        global_idle_op = _get_error_gate('idle', ideal_gate, depolarization_strengths,
+                                         stochastic_error_probs, lindblad_error_coeffs,
+                                         depolarization_parameterization, stochastic_parameterization,
+                                         lindblad_parameterization)  # 1-qubit idle op
     else:
-        idle_op = None
+        global_idle_op = None
 
     prep_layers = {}
-    if 'prep' in all_keys:
+    if evotype == 'chp':
+        assert ('prep' not in all_keys), "Cannot specify 'prep' error specification with CHP evotype"
+
+        # For CHP, prep can only be idle gate
+        rho0 = _op.ComposedOp([_op.EmbeddedOp(range(num_qubits), [i], _op.StaticStandardOp('Gi', 'chp'))
```

Comment:
It still seems incorrect to me to set 'rho0' to be an operation intstead of a spam vector.

Reply:
Agreed, but we decided to defer this issue for a more general SPAMVec + noisy op implementation in the near future. Comment added here to indicate that.